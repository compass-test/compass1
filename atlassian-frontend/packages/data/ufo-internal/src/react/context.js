import React, {
  Component,
  createContext,
  useContext,
  useEffect,
  useRef,
} from 'react';

import { v4 as uuidv4 } from 'uuid';

import {
  ConcurrentExperience,
  UFOExperience,
} from '@atlassian/ufo-experimental';
import { ufolog } from '@atlassian/ufo-experimental/logger';

export const getActions = () => {
  return {
    actions: {},
  };
};

export const UFOContext = createContext(null);

type ErrorBoundaryProps = {
  metric: UFOExperience,
};

class UFOErrorBoundary extends Component<ErrorBoundaryProps> {
  componentDidCatch(/*error, errorInfo*/) {
    this.props.metric.failure();
  }

  render() {
    return this.props.children;
  }
}

type ContextProps = {
  metric: UFOExperience,
  start?: boolean,
};

export const UFOContextComponent = (props: ContextProps) => {
  const context = useContext(UFOContext);

  const hasBeenCalled = useRef(false);
  const firstRun = hasBeenCalled.current === false;

  if (hasBeenCalled.current === false) {
    hasBeenCalled.current = uuidv4();
  }

  //console.log('context', props.metric.id, context);

  const metric =
    props.metric instanceof ConcurrentExperience
      ? props.metric.getInstance(hasBeenCalled.current)
      : props.metric;

  if (context) {
    //metric.setParent(context);
  }
  if ((props.start === undefined || props.start !== false) && firstRun) {
    ufolog('START', props.metric.id, hasBeenCalled.current);
    metric.start();
  }

  useEffect(() => {
    return () => {
      props.metric.release?.(hasBeenCalled.current);
    };
  }, [props.metric]);

  return (
    <UFOContext.Provider value={metric}>
      <UFOErrorBoundary metric={metric}>
        <div data-uuid={`${props.metric.id} ${hasBeenCalled.current}`}>
          {props.children}
        </div>
      </UFOErrorBoundary>
    </UFOContext.Provider>
  );
};
