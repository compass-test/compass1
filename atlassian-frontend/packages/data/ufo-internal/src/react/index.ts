export { useUFOComponentExperience } from './use-component-experience';
export { ExperienceSuccess } from './experience-success';
