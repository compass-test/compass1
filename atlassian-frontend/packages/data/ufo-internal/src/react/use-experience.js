// PLACEHOLDER for concurrent experiences for components
export const useExperience = experience => {
  return experience;
};
