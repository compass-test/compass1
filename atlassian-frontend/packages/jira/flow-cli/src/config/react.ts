export default {
  Component: 'React$Component',
  'React.Component': 'React$Component',
  RefObject: 'React$Ref',
  'React.RefObject': 'React$Ref',
  Ref: 'React$Ref',
  'React.Ref': 'React$Ref',
  ReactNode: 'React$Node',
  'React.ReactNode': 'React$Node',
  ElementType: 'React$ElementType',
  'React.ElementType': 'React$ElementType',
  ReactElement: 'React$Element',
  'React.ReactElement': 'React$Element',
  SFC: 'React$ComponentType',
  'React.SFC': 'React$ComponentType',
  PureComponent: 'React$Component',
  'React.PureComponent': 'React$Component',
  ComponentClass: 'React$ComponentType',
  'React.ComponentClass': 'React$ComponentType',
  ComponentType: 'React$ComponentType',
  'React.ComponentType': 'React$ComponentType',
  StatelessComponent: 'React$ComponentType',
  'React.StatelessComponent': 'React$ComponentType',
  MouseEvent: 'SyntheticMouseEvent',
  'React.MouseEvent': 'SyntheticMouseEvent',
  FocusEvent: 'SyntheticFocusEvent',
  'React.FocusEvent': 'SyntheticFocusEvent',
  KeyboardEvent: 'SyntheticKeyboardEvent',
  'React.KeyboardEvent': 'SyntheticKeyboardEvent',
  ReactPortal: 'React$Portal',
  'React.ReactPortal': 'React$Portal',
  ReactChild: 'React$Node',
  'React.ReactChild': 'React$Node',
  FC: 'React$ComponentType',
  'React.FC': 'React$ComponentType',
  Context: 'React$Context',
  'React.Context': 'React$Context',

  MouseEventHandler: '(event: SyntheticMouseEvent<any>) => void',
  'MouseEventHandler<HTMLElement>':
    '(event: SyntheticMouseEvent<HTMLElement>) => void',
  'MouseEventHandler<HTMLDivElement>':
    '(event: SyntheticMouseEvent<HTMLDivElement>) => void',
  'MouseEventHandler<HTMLButtonElement>':
    '(event: SyntheticMouseEvent<HTMLButtonElement>) => void',
  FocusEventHandler: '(event: SyntheticFocusEvent<any>) => void',
  'FocusEventHandler<HTMLElement>':
    '(event: SyntheticFocusEvent<HTMLElement>) => void',
  'FocusEventHandler<HTMLInputElement>':
    '(event: SyntheticFocusEvent<HTMLInputElement>) => void',
  'FocusEventHandler<HTMLTextAreaElement>':
    '(event: SyntheticFocusEvent<HTMLTextAreaElement>) => void',
  'FocusEventHandler<HTMLDivElement>':
    '(event: SyntheticFocusEvent<HTMLDivElement>) => void',
  'FocusEventHandler<HTMLButtonElement>':
    '(event: SyntheticFocusEvent<HTMLButtonElement>) => void',
  'React.MouseEventHandler': '(event: SyntheticFocusEvent<any>) => void',
  'React.MouseEventHandler<HTMLElement>':
    '(event: SyntheticMouseEvent<HTMLElement>) => void',
  'React.MouseEventHandler<HTMLDivElement>':
    '(event: SyntheticMouseEvent<HTMLDivElement>) => void',
  'React.MouseEventHandler<HTMLButtonElement>':
    '(event: SyntheticMouseEvent<HTMLButtonElement>) => void',
  'React.FocusEventHandler': '(event: SyntheticFocusEvent<any>) => void',
  'React.FocusEventHandler<HTMLElement>':
    '(event: SyntheticFocusEvent<HTMLElement>) => void',
  'React.FocusEventHandler<HTMLInputElement>':
    '(event: SyntheticFocusEvent<HTMLInputElement>) => void',
  'React.FocusEventHandler<HTMLTextAreaElement>':
    '(event: SyntheticFocusEvent<HTMLTextAreaElement>) => void',
  'React.FocusEventHandler<HTMLDivElement>':
    '(event: SyntheticFocusEvent<HTMLDivElement>) => void',
  'React.FocusEventHandler<HTMLButtonElement>':
    '(event: SyntheticFocusEvent<HTMLButtonElement>) => void',

  'PropTypes.Requireable<any>': 'any',
};
