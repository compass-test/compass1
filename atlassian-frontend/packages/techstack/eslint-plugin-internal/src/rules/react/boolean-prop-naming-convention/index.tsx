import type { Rule } from 'eslint';

const prefixes = ['is', 'has', 'should'];
const validTypes = ['TSBooleanKeyword', 'TSUndefinedKeyword', 'TSNullKeyword'];

const isUpperCase = (str: string) => {
  return str.toUpperCase() === str;
};

const rule: Rule.RuleModule = {
  meta: {
    fixable: 'code',
    messages: {
      propShouldBeCamelCase: 'Prop should be CamelCase',
      propNeedsPrefix: `Prop should be prefixed with one of the following: "${prefixes.join(
        ', ',
      )}".`,
    },
  },
  create(context) {
    return {
      'TSInterfaceDeclaration[id.name=/(Props|PropTypes)$/] TSPropertySignature,TSTypeAliasDeclaration[id.name=/(Props|PropTypes)$/] TSPropertySignature': (
        node: any,
      ) => {
        const name = node.key.value || node.key.name;
        if (name.includes('-')) {
          // Prop most likely is kebab-case - abort!
          return;
        }

        if (node.typeAnnotation) {
          if (
            !['TSBooleanKeyword', 'TSUnionType'].includes(
              node.typeAnnotation.typeAnnotation.type,
            )
          ) {
            // Node is not a boolean - abort!
            return;
          }

          if (node.typeAnnotation.typeAnnotation.types) {
            let hasBoolean = false;

            const hasInvalidTypes = node.typeAnnotation.typeAnnotation.types.some(
              (type: any) => {
                if (type.type === 'TSBooleanKeyword') {
                  hasBoolean = true;
                }

                if (validTypes.includes(type.type)) {
                  return false;
                }

                return true;
              },
            );

            if (hasInvalidTypes || !hasBoolean) {
              // Prop has no boolean type or has invalid types - abort!
              return;
            }
          }
        }

        for (let i = 0; i < prefixes.length; i++) {
          const prefix = prefixes[i];
          if (name.startsWith(prefix)) {
            // Prop is prefixed - let's make sure its CamelCase
            const nextCharIndex = name.indexOf(prefix) + prefix.length;
            const nextChar = name[nextCharIndex];
            if (isUpperCase(nextChar)) {
              // We're good!
              return;
            }

            // Next char isn't upper case - error!
            context.report({
              node,
              messageId: 'propShouldBeCamelCase',
              fix: fixer => {
                let fixedName = name.split('');
                fixedName[nextCharIndex] = name[nextCharIndex].toUpperCase();
                fixedName = fixedName.join('');
                return fixer.replaceText(node.key, fixedName);
              },
            });

            return;
          }
        }

        // Prop is not prefixed - time to error!
        context.report({
          node,
          messageId: 'propNeedsPrefix',
        });
      },
    };
  },
};

export default rule;
