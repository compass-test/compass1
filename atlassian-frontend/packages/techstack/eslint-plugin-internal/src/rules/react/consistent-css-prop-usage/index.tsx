import type { Rule, Scope } from 'eslint';
// eslint-disable-next-line import/no-unresolved
import type { CallExpression, Expression, Node, SpreadElement } from 'estree';

import { findIdentifierInParentScope } from '../../../utils/find-identifier-in-parent-scope';

const declarationSuffix = 'Styles';

function isCssCallExpression(node: Node): node is CallExpression {
  return !!(
    node.type === 'CallExpression' &&
    node.callee &&
    node.callee.type === 'Identifier' &&
    node.callee.name === 'css' &&
    node.arguments.length &&
    node.arguments[0].type === 'ObjectExpression'
  );
}

function findSpreadProperties(node: any): SpreadElement[] {
  return node.properties.filter(
    (property: any) =>
      property.type === 'SpreadElement' ||
      property.type === 'ExperimentalSpreadProperty',
  );
}

function analyzeIdentifier(
  context: Rule.RuleContext,
  sourceIdentifier: Scope.Reference['identifier'],
) {
  const scope = context.getScope();
  const identifier = findIdentifierInParentScope({
    scope,
    identifierName: sourceIdentifier.name,
  });
  if (!identifier || !identifier.parent) {
    // Identifier isn't in the module, skip!
    return;
  }

  if (identifier.parent.type !== 'VariableDeclarator') {
    // When variable is not in the file or coming from import
    context.report({
      node: sourceIdentifier,
      messageId: 'cssInModule',
    });
    return;
  }

  if (identifier.parent.parent.parent.type !== 'Program') {
    // When variable is declared inside the component
    context.report({
      node: sourceIdentifier,
      messageId: 'cssOnTopOfModule',
    });
    return;
  }

  if (
    identifier.parent &&
    identifier.parent.init &&
    !isCssCallExpression(identifier.parent.init)
  ) {
    // When variable value is not of type css({})
    context.report({
      node: identifier,
      messageId: 'cssObjectTypeOnly',
    });
    return;
  }

  const spreadProperties =
    identifier.parent.init?.type === 'CallExpression' &&
    findSpreadProperties(identifier.parent.init?.arguments[0]);
  if (spreadProperties) {
    // TODO: Recursively handle spread items in children properties.
    spreadProperties.forEach((prop: any) => {
      context.report({
        node: prop,
        messageId: 'cssArrayStylesOnly',
      });
    });
  }
}

function traverseExpression(
  context: Rule.RuleContext,
  expression: Expression | SpreadElement,
) {
  switch (expression.type) {
    case 'Identifier':
      // {styles}
      // We've found an identifier - time to analyze it!
      analyzeIdentifier(context, expression);
      break;

    case 'ArrayExpression':
      // {[styles, moreStyles]}
      // We've found an array expression - let's traverse again over each element individually.
      expression.elements.forEach(element =>
        traverseExpression(context, element),
      );
      break;

    case 'LogicalExpression':
      // {isEnabled && styles}
      // We've found a logical expression - we're only interested in the right expression so
      // let's traverse that and see what it is!
      traverseExpression(context, expression.right);
      break;

    case 'ConditionalExpression':
      // {isEnabled ? styles : null}
      // We've found a conditional expression - we're only interested in the consequent and
      // alternate (styles : null)
      traverseExpression(context, expression.consequent);
      traverseExpression(context, expression.alternate);
      break;

    case 'CallExpression':
    case 'ObjectExpression':
    case 'TaggedTemplateExpression':
    case 'TemplateLiteral':
      // We've found elements that shouldn't be here! Report an error.
      context.report({
        node: expression,
        messageId: 'cssOnTopOfModule',
      });
      break;

    default:
      // Do nothing!
      break;
  }
}

const rule: Rule.RuleModule = {
  meta: {
    fixable: 'code',
    messages: {
      cssOnTopOfModule: `Styles should be defined at the top of the module using the \`css({})\` function.`,
      cssObjectTypeOnly: `Styles should be created using the \`css({})\` function.`,
      cssInModule: `Imported styles are not allowed to be used, define in the module, import a component, or use a theme token instead.`,
      cssArrayStylesOnly: `Combine styles using an array on the CSS prop instead of object spread, e.g. \`css([containerStyle, backgroundStyle])\`.`,
      shouldEndInStyles: 'Variable should end in "styles".',
    },
  },
  create(context) {
    return {
      'CallExpression[callee.name=css]': (node: Rule.Node) => {
        if (node.parent.type !== 'VariableDeclarator') {
          // We aren't interested in these that don't have a parent.
          return;
        }

        const identifier = node.parent.id;
        if (
          identifier.type === 'Identifier' &&
          identifier.name.endsWith(declarationSuffix)
        ) {
          // Already prefixed! Nothing to do.
          return;
        }

        context.report({
          node: identifier,
          messageId: 'shouldEndInStyles',
          fix: fixer => {
            const identifierName =
              identifier.type === 'Identifier' ? identifier.name : '';
            const references: Scope.Scope['references'] =
              context
                .getScope()
                .variables.find(varb => varb.name === identifierName)
                ?.references || [];

            const newIdentifierName = `${identifierName
              // Remove "Style" if it is already defined.
              .replace(/Style$/, '')}${declarationSuffix}`;

            return references
              .filter(ref => ref.identifier.name === identifierName)
              .map((ref: any) => {
                if (ref.identifier.parent && ref.identifier.parent.shorthand) {
                  return fixer.replaceText(
                    ref.identifier,
                    `${identifierName}: ${newIdentifierName}`,
                  );
                }

                return fixer.replaceText(ref.identifier, newIdentifierName);
              });
          },
        });
      },
      JSXAttribute(node: any) {
        const { name, value } = node;

        if (name.type === 'JSXIdentifier' && name.name === 'css') {
          // When not a jsx expression. For eg. css=""
          if (value.type !== 'JSXExpressionContainer') {
            context.report({
              node: value,
              messageId: 'cssOnTopOfModule',
            });

            return;
          }

          traverseExpression(context, value.expression);
        }
      },
    };
  },
};

export default rule;
