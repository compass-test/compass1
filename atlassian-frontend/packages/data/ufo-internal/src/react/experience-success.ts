import { useLayoutEffect } from 'react';

import { UFOExperience } from '@atlassian/ufo-experimental';

type Props = { experience: UFOExperience };

/**
 * A React component that succeeds the desired experience when mounted. Uses useLayoutEffect to ensure accuracy of success measurement
 * Use this instead of a straight call to `experience.success()` if your success point is during a render.
 * @param experience
 * @constructor
 */
export const ExperienceSuccess = ({ experience }: Props) => {
  useLayoutEffect(() => {
    experience.success();
  }, [experience]);
  return null;
};
