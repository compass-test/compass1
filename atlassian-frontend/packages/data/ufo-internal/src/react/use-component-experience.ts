import { useEffect, useRef } from 'react';

import { UFOExperience } from '@atlassian/ufo-experimental';

const useConstructor = (callback: () => void) => {
  const hasBeenFired = useRef(false);
  if (!hasBeenFired.current) {
    callback();
    hasBeenFired.current = true;
  }
  return null;
};

/**
 * A hook to start an experience and to auto abort the experience when the parent component is unmounted.
 * Use this instead of a direct call to `experience.start`. If you need to restart the experience
 * simply trigger an unmount and remount of the parent component.
 * @param experience the experience to start and abort
 */
export const useUFOComponentExperience = (experience: UFOExperience) => {
  useEffect(() => {
    return () => {
      experience.abort();
    };
    // we want this cleanup to only happen on unmount so this is a legit use of empty array
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useConstructor(() => experience.start());
  return null;
};
