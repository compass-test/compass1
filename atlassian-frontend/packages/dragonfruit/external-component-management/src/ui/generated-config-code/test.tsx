import React from 'react';

import { act, render, RenderResult } from '@testing-library/react';
import fileDownload from 'js-file-download';

import {
  CompassComponentDetailViewFragment,
  CompassComponentType,
  CompassLinkType,
  CompassRelationshipType,
} from '@atlassian/dragonfruit-graphql';
import { CompassTestProvider } from '@atlassian/dragonfruit-testing';

import {
  CONFIG_AS_CODE_DAC_LINK,
  CONFIG_AS_CODE_FILE_NAME,
} from '../../constants';

import GeneratedConfigCode from './main';

jest.mock('js-file-download', () => ({
  __esModule: true,
  default: jest.fn(),
}));

const baseComponent: CompassComponentDetailViewFragment = {
  name: 'TestName',
  id:
    'ari:cloud:compass:4958bb5d-3970-4a13-bebc-62bbca57f370:component/5ce8c075-7b72-4455-9be9-7f0a1c6e6db4/a60d9b0f-fa86-436c-b3c2-0eece8e94fdb',
  type: CompassComponentType.SERVICE,
};

const baseYaml = `name: ${baseComponent.name}
id: '${baseComponent.id}'
description: null
ownerId: null
fields: null
links: null
relationships: {}

# Learn more about formatting compass.yml:
# ${CONFIG_AS_CODE_DAC_LINK}`;

const componentWithTier: CompassComponentDetailViewFragment = {
  name: 'TestName',
  id:
    'ari:cloud:compass:d7192f2f-11f8-4c7c-950c-af5ad04b12f8:component/79fc4a02-929a-43e6-93f5-3d4d9d54e942/0e8db7aa-4624-4f09-b51c-03b251b859bc',
  type: CompassComponentType.SERVICE,
  fields: [
    {
      definition: {
        id: 'compass:tier',
      },
      value: ['4'],
    },
  ],
};

const codeblockTestId = 'dragonfruit.config-as-code-generated-codeblock';

describe('GeneratedConfigCode', () => {
  test('renders successfully  ', () => {
    const { getByTestId } = render(
      <CompassTestProvider>
        <GeneratedConfigCode component={baseComponent} />
      </CompassTestProvider>,
    );

    expect(getByTestId(codeblockTestId)).toBeDefined();
  });

  describe('YAML generation', () => {
    test('only required fields', () => {
      const { getByTestId } = render(
        <CompassTestProvider>
          <GeneratedConfigCode component={baseComponent} />
        </CompassTestProvider>,
      );
      const generatedCode = getByTestId(codeblockTestId).innerText;
      expect(generatedCode).toMatchSnapshot();
    });

    test('all possible fields', () => {
      const component = {
        ...baseComponent,
        ownerId: 'someownerid',
        description: 'description goes here',
        links: [
          {
            name: 'Link 1',
            id: 'link1id',
            type: CompassLinkType.DASHBOARD,
            url: 'http://www.atlassian.net',
          },
          {
            name: 'Link 2',
            id: 'link2id',
            type: CompassLinkType.DOCUMENT,
            url: 'http://www.atlassian.net/wiki',
          },
        ],
        relationships: {
          nodes: [
            {
              type: CompassRelationshipType.DEPENDS_ON,
              startNode: baseComponent,
              endNode: { id: 'abc123' },
            },
          ],
        },
      };

      const { getByTestId } = render(
        <CompassTestProvider>
          <GeneratedConfigCode component={component} />
        </CompassTestProvider>,
      );

      const generatedCode = getByTestId(codeblockTestId).innerText;
      expect(generatedCode).toMatchSnapshot();
    });
    test('component with tier', () => {
      const { getByTestId } = render(
        <CompassTestProvider>
          <GeneratedConfigCode component={componentWithTier} />
        </CompassTestProvider>,
      );

      const generatedCode = getByTestId(codeblockTestId).innerText;
      expect(generatedCode).toMatchSnapshot();
    });
  });

  describe('copy and download', () => {
    let wrapper: RenderResult;

    beforeEach(() => {
      wrapper = render(
        <CompassTestProvider>
          <GeneratedConfigCode component={baseComponent} />
        </CompassTestProvider>,
      );
    });

    test('can copy config file', () => {
      const writeTextMock = jest.fn();
      Object.assign(navigator, {
        clipboard: {
          writeText: writeTextMock,
        },
      });

      const copyButton = wrapper.getByTestId(
        'dragonfruit.config-as-code-setup-modal.copy-button',
      );
      act(() => copyButton.click());

      expect(writeTextMock).toBeCalled();
      expect(writeTextMock).toBeCalledWith(baseYaml);
    });

    test('can download config file', () => {
      const downloadButton = wrapper.getByTestId(
        'dragonfruit.config-as-code-setup.download-button',
      );
      act(() => downloadButton.click());

      expect(fileDownload).toBeCalled();
      expect(fileDownload).toBeCalledWith(baseYaml, CONFIG_AS_CODE_FILE_NAME);
    });
  });
});
