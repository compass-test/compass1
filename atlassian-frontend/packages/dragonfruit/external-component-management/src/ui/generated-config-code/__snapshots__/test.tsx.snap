// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneratedConfigCode YAML generation all possible fields 1`] = `
"1name: TestName
2id: 'ari:cloud:compass:4958bb5d-3970-4a13-bebc-62bbca57f370:component/5ce8c075-7b72-4455-9be9-7f0a1c6e6db4/a60d9b0f-fa86-436c-b3c2-0eece8e94fdb'
3description: description goes here
4ownerId: someownerid
5fields: null
6links:
7  - name: Link 1
8    type: DASHBOARD
9    url: 'http://www.atlassian.net'
10  - name: Link 2
11    type: DOCUMENT
12    url: 'http://www.atlassian.net/wiki'
13relationships:
14  DEPENDS_ON:
15    - abc123
16
17# Learn more about formatting compass.yml:
18# https://go.atlassian.com/compass-config-as-code"
`;

exports[`GeneratedConfigCode YAML generation component with tier 1`] = `
"1name: TestName
2id: 'ari:cloud:compass:d7192f2f-11f8-4c7c-950c-af5ad04b12f8:component/79fc4a02-929a-43e6-93f5-3d4d9d54e942/0e8db7aa-4624-4f09-b51c-03b251b859bc'
3description: null
4ownerId: null
5fields:
6  tier: 4
7links: null
8relationships: {}
9
10# Learn more about formatting compass.yml:
11# https://go.atlassian.com/compass-config-as-code"
`;

exports[`GeneratedConfigCode YAML generation only required fields 1`] = `
"1name: TestName
2id: 'ari:cloud:compass:4958bb5d-3970-4a13-bebc-62bbca57f370:component/5ce8c075-7b72-4455-9be9-7f0a1c6e6db4/a60d9b0f-fa86-436c-b3c2-0eece8e94fdb'
3description: null
4ownerId: null
5fields: null
6links: null
7relationships: {}
8
9# Learn more about formatting compass.yml:
10# https://go.atlassian.com/compass-config-as-code"
`;
